# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/nsCSPUtils.cpp
# Commit: 23ca1773d73a
# Full Hash: 23ca1773d73a13c012b1cc635fe6615da46a4675
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2025-01-17 21:57:54
# Regressor Bug: 1901492
# File Overlap Count: 1
# Description:
#   Bug 1901492 - Simplify some require-trusted-types-for checks. r=smaug
#   
#   Currently, there is only one sink group defined in the spec, accepted
#   by nsCSPParser::handleRequireTrustedTypesForDirective and used by
#   callers of GetTrustedTypesCompliantString. This means that the flag
# ==============================================================================

diff -r 6d76c1674ad7 -r 23ca1773d73a dom/security/nsCSPUtils.cpp
--- a/dom/security/nsCSPUtils.cpp	Fri Jan 17 13:31:12 2025 +0000
+++ b/dom/security/nsCSPUtils.cpp	Fri Jan 17 13:35:39 2025 +0000
@@ -29,6 +29,7 @@
 #include "mozilla/dom/CSPDictionariesBinding.h"
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/SRIMetadata.h"
+#include "mozilla/dom/TrustedTypesConstants.h"
 #include "mozilla/StaticPrefs_security.h"
 
 using namespace mozilla;
@@ -1061,11 +1062,6 @@
   aOutStr.Append(mValue);
 }
 
-bool nsCSPRequireTrustedTypesForDirectiveValue::
-    isRequiresTrustedTypesForSinkGroup(const nsAString& aSinkGroup) const {
-  return mValue == aSinkGroup;
-}
-
 /* =============== nsCSPTrustedTypesDirectivePolicyName =============== */
 
 nsCSPTrustedTypesDirectivePolicyName::nsCSPTrustedTypesDirectivePolicyName(
@@ -1398,15 +1394,6 @@
   return false;
 }
 
-bool nsCSPDirective::AreTrustedTypesForSinkGroupRequired(
-    const nsAString& aSinkGroup) const {
-  MOZ_ASSERT(mDirective ==
-             nsIContentSecurityPolicy::REQUIRE_TRUSTED_TYPES_FOR_DIRECTIVE);
-
-  return mSrcs.Length() == 1 &&
-         mSrcs[0]->isRequiresTrustedTypesForSinkGroup(aSinkGroup);
-}
-
 void nsCSPDirective::toString(nsAString& outStr) const {
   // Append directive name
   outStr.AppendASCII(CSP_CSPDirectiveToString(mDirective));
@@ -1857,9 +1844,6 @@
 }
 
 bool nsCSPPolicy::hasDirective(CSPDirective aDir) const {
-  if (aDir == nsIContentSecurityPolicy::REQUIRE_TRUSTED_TYPES_FOR_DIRECTIVE) {
-    return mHasRequireTrustedTypesForDirective;
-  }
   for (uint32_t i = 0; i < mDirectives.Length(); i++) {
     if (mDirectives[i]->equals(aDir)) {
       return true;
@@ -1894,18 +1878,8 @@
 
 bool nsCSPPolicy::AreTrustedTypesForSinkGroupRequired(
     const nsAString& aSinkGroup) const {
-  if (!hasDirective(
-          nsIContentSecurityPolicy::REQUIRE_TRUSTED_TYPES_FOR_DIRECTIVE)) {
-    return false;
-  }
-  for (const auto* directive : mDirectives) {
-    if (directive->equals(
-            nsIContentSecurityPolicy::REQUIRE_TRUSTED_TYPES_FOR_DIRECTIVE)) {
-      return directive->AreTrustedTypesForSinkGroupRequired(aSinkGroup);
-    }
-  }
-
-  return false;
+  MOZ_ASSERT(aSinkGroup == kTrustedTypesOnlySinkGroup);
+  return mHasRequireTrustedTypesForDirective;
 }
 
 /*